lib/block_view_stake.go

```diff
@@ -2137,6 +2137,40 @@ func (bav *UtxoView) IsValidUnlockStakeMetadata(transactorPkBytes []byte, metada
 	return nil
 }
 
+// IsCorrectValidatorTotalStakeAmountNanos returns true if the total stake amount for the validator
+// matches the total stake amount calculated from the StakeEntries in the UtxoView + DB and false otherwise.
+func (bav *UtxoView) IsCorrectValidatorTotalStakeAmountNanos(validatorEntry *ValidatorEntry) (bool, error) {
+	// make a copy of the stake entry map to use here, so we don't have to flush a bunch of entries that don't change
+	// after we merge the DB results in.
+	stakeEntryMapCopy := make(map[StakeMapKey]*StakeEntry, len(bav.StakeMapKeyToStakeEntry))
+	for stakeMapKey, stakeEntry := range bav.StakeMapKeyToStakeEntry {
+		stakeEntryMapCopy[stakeMapKey] = stakeEntry.Copy()
+	}
+	stakeEntries, err := DBGetStakeEntriesForValidatorPKID(bav.Handle, bav.Snapshot, validatorEntry.ValidatorPKID)
+	if err != nil {
+		return false, errors.Wrapf(err, "IsCorrectValidatorTotalStakeAmountNanos: error retrieving StakeEntries: ")
+	}
```

Comment:
I think we want to merge the entries from the UtxoView and Badger in the opposite order, no? I.e. first pull the entries from badger, then apply all changes from the view. This way, change in the view aren't overwritten by stale data from Badger

Reply:
It works as currently written as I skip entries if they already exist in the map. However, I can write it the other way around if it makes it more clear.